aws/resource_aws_lambda_layer_test.go

```diff
@@ -0,0 +1,215 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/hashicorp/terraform/terraform"
+
+	"github.com/aws/aws-sdk-go/service/lambda"
+	"github.com/hashicorp/terraform/helper/resource"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+)
+
+func init() {
+	resource.AddTestSweepers("aws_lambda_layer", &resource.Sweeper{
+		Name: "aws_lambda_layer",
+		F:    testSweepLambdaLayers,
+	})
+}
+
+func testSweepLambdaLayers(region string) error {
+	client, err := sharedClientForRegion(region)
+	if err != nil {
+		return fmt.Errorf("error getting client: %s", err)
+	}
+
+	lambdaconn := client.(*AWSClient).lambdaconn
+	resp, err := lambdaconn.ListLayers(&lambda.ListLayersInput{})
+	if err != nil {
+		if testSweepSkipSweepError(err) {
+			log.Printf("[WARN] Skipping Lambda Layer sweep for %s: %s", region, err)
+			return nil
+		}
+		return fmt.Errorf("Error retrieving Lambda layers: %s", err)
+	}
+
+	if len(resp.Layers) == 0 {
+		log.Print("[DEBUG] No aws lambda layers to sweep")
+		return nil
+	}
+
+	for _, l := range resp.Layers {
+		var testOptGroup bool
+		for _, testName := range []string{"tf_test", "tf_acc_"} {
```

Comment:
Since we are starting fresh, we should be able to stick to a consistent naming scheme and shouldn't need to loop through a list of prefixes. 👍